pkg/controllers/hostedcluster/hostedcluster.go

```diff
@@ -0,0 +1,76 @@
+package hostedcluster
+
+import (
+	hyperv1beta1 "github.com/openshift/hypershift/api/v1beta1"
+
+	// _ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	aggregatorapiserver "k8s.io/kube-aggregator/pkg/apiserver"
+	operatorclient "open-cluster-management.io/api/client/operator/clientset/versioned"
+	"open-cluster-management.io/multicluster-controlplane/pkg/controllers/ocmcontroller"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/stolostron/multicluster-controlplane/pkg/controllers/hostedcluster/controller"
+)
+
+var scheme = runtime.NewScheme()
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(hyperv1beta1.AddToScheme(scheme))
+}
+
+// InstallControllers installs next-gen controlplane controllers in hub cluster
+func InstallControllers(stopCh <-chan struct{}, aggregatorConfig *aggregatorapiserver.Config) error {
+	ctx := ocmcontroller.GoContext(stopCh)
+
+	loopbackRestConfig := aggregatorConfig.GenericConfig.LoopbackClientConfig
+	loopbackRestConfig.ContentType = "application/json"
+	controlplaneKubeClient, err := kubernetes.NewForConfig(loopbackRestConfig)
+	if err != nil {
+		return err
+	}
+	controlplaneOperatorClient, err := operatorclient.NewForConfig(loopbackRestConfig)
+	if err != nil {
+		return err
+	}
+
+	restConfig, err := rest.InClusterConfig()
+	if err != nil {
```

Comment:
we support to run the controlplane as a binary, so if the controlplane is not running in a cluster, we can give a warning and just return directly